### DISCUSSION TITLE
Adding configuration management?

### DISCUSSION THREAD

**Discussion by wtremmel** [napalm-automation-community/napalm-ros#111] [Ideas]
Hi,

is there any chance to add configuration management to napalm-ros?
I really would like to configure Mikrotik routers using [Peering-Manager](https://peering-manager.net)...

Wolfgang

**Comment by luqasz** [comment]
There are two possible solutions.
### Script
1. Parse config from `/export`
2. Compare it with running config
3. Build mikrotik script that will apply only what is required 
4. Upload script and execute it

### API only modification
1. Parse e.g. json with required config
2. Prepare diff
3. Apply changes over API

For me solution with API is much simpler as it requires only parsing config and doing all necessary work through API. One downside is that it won't prevent any disconnections in middle of applying. 

Also please read #71
